### ISSUE TITLE
[CLI]: wand.log doesn't seem to work when including step=

### ISSUE THREAD

**Issue by grahamannett** [wandb/wandb#5274]
### Describe the bug

if i wandb.log it does not seem to upload when i use step argument as well.  i kind of believe it is something i am doing wrong because it seems like something others would have mentioned and i dont see any other issues or info in release note that is similar from a quick search.  
an example below, if there is a better way to do this (such that step on the graphs doesnt have large "jumps" from point to point) i would love to know but didnt seem to find anything in docs that didnt require multiple wandb.runs at once?

example below but if i use the line `wandb.log({"eval/loss": running_loss}, step=i, commit=True, sync=True)` (and have tried without sync=True or commit=True etc but including here to be more explicit) it does not seem to upload this metric/data as i cannot see it on browser or through api.

```python
import wandb

wandb.init(project="wanb-test-log-wahoo", config={"a": 1, "b": 2})

num_epochs = 100
num_batches = 1000
eval_every = 1
log_every = 10

running_loss = 1000
eval_metric = 1.0

for i in range(num_epochs):
    for ii in range(num_batches):
        running_loss *= 0.99
        if ii % log_every == 0:
            wandb.log({"train/loss_batch": running_loss})

    if i % eval_every == 0:
        eval_metric += 1 / num_epochs
        wandb.log({"eval/loss": running_loss}, step=i, commit=True, sync=True)

    wandb.log({"train/epoch": i, "train/loss": i / running_loss})


```

### Environment

WandB version: 0.14.0

Python version: 3.10

**Comment by kptkin** [comment]
@grahamannett Thanks for writing in. Few points:
1. `sync` is deprecated and it doesn't change the behavior of `log`.
2. Now, if you look in the logs of the run (you can find it under the folder you ran your the script from, the path would be something like this: `/wandb/run-<date>_<timestemp>-<run_id>/logs/debug-internal.log`) you will find warning that says that we drop all the `"eval/loss"` values, search for ` Dropping entry: {'eval/loss'...`. 
     - Why? 
         - We currently only support logging monotonically increasing steps. Every time you do ` wandb.log({"train/loss_batch": running_loss})` without providing step we increment the internal step by 1 so when we get to the `wandb.log({"eval/loss": running_loss}, step=i, commit=True, sync=True)` the step is way ahead of `i` and we drop this entry.
     - How to fix? 
          - If you want to log against a custom value for your step you could try using `define_metric`. See [docs](https://docs.wandb.ai/guides/track/log/customize-logging-axes#customize-axes) for more details on usage.
          - If you want to use the same global step you need to make sure the values for the `step` are monotonically increasing.

Hope this helps, please let us know if you have additional questions.

**Comment by MovingKyu** [comment]
I have some problems with logging on "same step". I run both validation and test dataset for evaluation in the same training step, and I log them individually. If I input same "step" number to both test and validation score (i.e. accuracy), wandb only logs either one of them. Now I avoid this issue by incrementing "step" by 1 when passing the value to log test scores. Do you have any idea why this happens?
